AspNetCoreApplication/Extensions/ClaimsPrincipalExtensions.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace AspNetCoreApplication.Extensions
+{
+    public static class ClaimsPrincipalExtensions
+    {
+        public static string GetUsername(this ClaimsPrincipal user)
+        {
+            return user.FindFirst(ClaimTypes.Upn)?.Value;
+        }
+
+        public static string GetUserId(this ClaimsPrincipal user)
+        {
+            return user.FindFirst(ClaimTypes.NameIdentifier)?.Value;
+        }
+        public static string GetName(this ClaimsPrincipal user)
+        {
+            return user.FindFirst(ClaimTypes.GivenName)?.Value;
+        }
+        public static string GetRole(this ClaimsPrincipal user)
+        {
+            return user.FindFirst(ClaimTypes.Role)?.Value;
+        }
```

Comment:
Chỉ nên khai báo 1 hàm thôi
```suggestion
        public static string GetClaimValue(this ClaimsPrincipal user, string claimType)
        {
            return user.FindFirst(claimType)?.Value;
        }
```

Reply:
done